packages/app/src/cli/services/generate-schema.ts

```diff
@@ -0,0 +1,30 @@
+import {session, output, api, store} from '@shopify/cli-kit'
+import {AppInterface} from '../models/app/app.js'
+import {FunctionExtension} from '../models/app/extensions.js'
+import {getAppIdentifiers} from '../models/app/identifiers.js'
+import {fetchOrganizationAndFetchOrCreateApp} from './environment.js'
+
+interface GenerateSchemaOptions {
+  app: AppInterface
+  extension: FunctionExtension
+  apiKey?: string
+}
+
+export async function generateSchemaService(options: GenerateSchemaOptions) {
+  const {extension, app} = options
+  const token = await session.ensureAuthenticatedPartners()
+  const {apiVersion: version, type} = extension.configuration
+  const apiKey =
+    options.apiKey ||
+    getAppIdentifiers({app}).app ||
+    (await fetchOrganizationAndFetchOrCreateApp(app, token)).partnersApp.apiKey
+
+  const query = api.graphql.ApiSchemaDefinitionQuery
+  const variables: api.graphql.ApiSchemaDefinitionQueryVariables = {
+    apiKey,
+    version,
+    type,
+  }
+  const response: api.graphql.ApiSchemaDefinitionQuerySchema = await api.partners.request(query, token, variables)
+  output.info(response.definition ?? '')
```

Comment:
What does `""` mean? When can that happen? I'd recommend to output something more meaningful to the user. Otherwise they'll end up opening an issue reporting that they get no output.